City-of-Helsinki/youth-membership-admin-ui
# YM-381 | Register youth e2e

## Description

Added `e2e` test for registering new youth. Tests check that all fields that are marked as required throw validation error. In next step form is filled and every button except "Make thing x primary" are tested. Finally form is submitted and redirected to profile details page.

## Context

In addition to tests, I created `.testcaferc.json` file. This was necessary to alter the order in which testcafe runs the tests, as one would expect, it uses alphabetical order as default. Moving file order to different file should also keep package.json script section little cleaner. I decided to leave browser information inside browser-test scripts for now.

[YM-381](https://helsinkisolutionoffice.atlassian.net/browse/YM-381)

## How Has This Been Tested?
`yarn browser-test` => all tests pass.

## Manual Testing Instructions for Reviewers
1) Add values to `.env ` file (if you don't have these you can contact me):
- `REACT_APP_TESTING_USERNAME`
- `REACT_APP_TESTING_USERNAME_NO_ACCESS`
- `REACT_APP_TESTING_PASSWORD`

2) Login to [profile-admin](https://profiili-api.test.kuva.hel.ninja/admin/) and make sure "Ulla User" has no profile
3) Start local application `yarn start`
4) Run `yarn browser-test`
5) Delete the created profile to avoid unnecessary problems.
